webapps/frontend/app/viewmodels/containers/albanode.js

```diff
@@ -179,19 +179,19 @@ define([
         };
 
         self.generateEmptySlot = function() {
-            self.emptySlotMessage(undefined);
-                api.post('alba/nodes/' + self.guid() + '/generate_empty_slot')
-                    .done(function (data) {
-                        if (![undefined, null].contains(data)) {
-                            var slotID = Object.keys(data)[0];
-                            var slot = new Slot(slotID, self, self.albaBackend);
-                            slot.fillData(data[slotID]);
-                            self.slots.push(slot);
-                        }
-                    })
-                    .fail(function() {
-                        self.emptySlotMessage('Unable to request an empty slot.');
-                    });
+            api.post('alba/nodes/' + self.guid() + '/generate_empty_slot')
+                .done(function (data) {
+                    if (![undefined, null].contains(data)) {
```

Comment:
In case data would be `null` or `undefined` the previous `emptySlotMessage` will not be cleared. This is unwanted behavior. Instead i'd create an `else` stack and fill out `emptySlotMessage` with something in the lines of `Failed to create a new empty slot.`